lib/adf_diag.py

```diff
@@ -1080,3 +1108,227 @@ def derive_variables(self, vars_to_derive=None, ts_dir=None, overwrite=None):
+
+        # import needed standard modules:
+        import json
+
+        test = False  # True (copy files but don't run), False (copy files and run MDTF).  Could make this a yaml variable
+        verbose = 2  # 0 errors and major sections, 1 warnings, 2 detailed, 3 excessive
+
+        print("\n  Setting up MDTF...")
+        # We want access to the entire dict of mdtf_info
+        mdtf_info = self.__mdtf_info
+
+        #
+        # Create a dict with all the case info needed for MDTF case_list
+        #     Note that model and convention are hard-coded to CESM because that's all we expect here
+        #     This could be changed by inputing them into ADF with other MDTF-specific variables
+        #
+        case_list_keys = ["CASENAME", "FIRSTYR", "LASTYR", "model", "convention"]
+
+        # Casenames, paths and start/end years come through the ADF
+        case_names = self.get_cam_info("cam_case_name", required=True)
+        start_years = self.climo_yrs["syears"]
+        end_years = self.climo_yrs["eyears"]
+
+        case_list_all = list(())
+        for icase, case in enumerate(case_names):
+            case_list_values = [
+                case,
+                start_years[icase],
+                end_years[icase],
+                "CESM",
+                "CESM",
+            ]
+            case_list_all.append(dict(zip(case_list_keys, case_list_values)))
+        mdtf_info["case_list"] = (
+            case_list_all  # this list of dicts is the format wanted by MDTF
+        )
+
+        # The plot_path is given by case in ADF but MDTF needs one top dir, so use case 0
+        # Working dir and output dir can be different. These could be set in config.yaml
+        # but then we don't get the nicely formated plot_location
+        case_idx = 0
+        plot_path = self.plot_location[case_idx] + "/mdtf"
+        for var in ["WORKING_DIR", "OUTPUT_DIR"]:
+            if mdtf_info[var] == "default":
+                mdtf_info[var] = plot_path
+
+        #
+        # Write the input settings json file
+        #
+        mdtf_input_settings_filename = self.get_mdtf_info(
+            "mdtf_input_settings_filename", required=True
+        )
+
+        with open(
+            mdtf_input_settings_filename,
+            "w",
+            encoding="utf-8",
+        ) as out_file:
+            json.dump(mdtf_info, out_file, sort_keys=True, indent=4, ensure_ascii=False)
+        mdtf_codebase = self.get_mdtf_info("mdtf_codebase_loc")
+        print(f"\t Using MDTF code base {mdtf_codebase}")
+
+        #
+        # Move the data to the dir structure and file names expected by the MDTF
+        #    model_input_data/case/freq/case.VAR.freq.nc
+
+        self.move_tsfiles_for_mdtf(verbose)
+
+        #
+        # Submit the MDTF script in background mode, send output to mdtf.out file
+        #
+        mdtf_log = os.path.join(
+            "./mdtf.out"
+        )  # maybe set this to cam_diag_plot_loc: /glade/scratch/${user}/ADF/plots
```

Comment:
If `mdtf.out` is being written to the current local directory then this can just be:

```suggestion
        mdtf_log = "mdtf.out" # maybe set this to cam_diag_plot_loc: /glade/scratch/${user}/ADF/plots
```